src/de.rs

```diff
@@ -396,12 +397,16 @@ impl_integers! {
     i16,
     i32,
     i64,
+    i128,
     isize,
     u8,
     u16,
     u32,
     u64,
+    // TODO cannot support u128 as it is constrained type by default and current constraints uses i128 for bounds
+    // u128,
```

Comment:
Can we implement this with its signed variant being BigInt?

Reply:
Could you clarify this bit more?

Casting u128 to signed bytes indeed currently changes the meaning. We could use BigInt there. 

This particular line was more related to how constraints work currently, which uses `i128` for bounds, in here: https://github.com/librasn/rasn/blob/0aa23e3ac906f2f3917609101b8d104268f57ab1/src/types.rs#L270-L296

Maybe we could change the upper constraint bound to be `u128` type. That makes checking the bounds more challenging as more type conversions are included but it should be possible.

Reply:
Actually I tried to change the constraint behavior, and it does not seem to be easy.

E.g. if we add more types for
```rust
#[derive(Clone, Copy, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Hash)]
pub enum Bounded<S, E> {
    #[default]
    None,
    Single(S),
    Range {
        start: Option<S>,
        end: Option<E>,
    },
}
```

We would need to re-adjust the constraint logic a lot, because many functions currently return single type, and expect that the type is same for Single or Range variants, for example.